test/integration/requester-spec/headers.test.js

```diff
@@ -0,0 +1,236 @@
+                responseHeader = _.get(JSON.parse(response).headers, 'postman-token'),
+
+                requestHeaders = request.header.filter(function(header) {
+                    return (header.key === 'Postman-Token');
+                });
+
+            expect(regex.test(responseHeader)).to.be.true;
+            expect(regex.test(requestHeaders[0].value)).to.be.true;
+        });
+    });
+
+    describe('are set to false', function() {
+        before(function(done) {
+            this.run({
+                requester: {
+                    cacheControlHeader: false,
+                    postmanTokenHeader: false
+                },
+                collection: {
+                    item: [{
+                        name: 'Cache-Control and postman-token',
+                        request: {
+                            url: HOST
+                        }
+                    }]
+                }
+            }, function (err, results) {
+                testrun = results;
+                done(err);
+            });
+        });
+
+        it('should complete the run', function () {
+            expect(testrun).to.be.ok;
+            sinon.assert.calledOnce(testrun.start);
+            sinon.assert.calledOnce(testrun.done);
+            sinon.assert.calledWith(testrun.done.getCall(0), null);
+
+            sinon.assert.calledOnce(testrun.request);
+            sinon.assert.calledWith(testrun.request.getCall(0), null);
+
+            sinon.assert.calledOnce(testrun.response);
+            sinon.assert.calledWith(testrun.response.getCall(0), null);
+        });
+
+        it('should send request without `Cache-Control: no-cache` header', function () {
+            var request = testrun.request.getCall(0).args[3].toJSON(),
+                response = JSON.parse(testrun.response.getCall(0).args[2].stream.toString());
+
+            expect(request.header).to.deep.not.include({key: 'Cache-Control', value: 'no-cache'});
+            expect(response.headers).to.not.have.property('cache-control', 'no-cache');
+        });
+
+        it('should send request without `Postman-Token` header', function () {
+            var request = testrun.request.getCall(0).args[3].toJSON(),
+                response = testrun.response.getCall(0).args[2].stream.toString(),
+
+                regex = /^(Postman-Token: )?[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/mi,
+
+                responseHeader = _.get(JSON.parse(response).headers, 'postman-token'),
+
+                requestHeaders = request.header.filter(function(header) {
+                    return (header.key === 'Postman-Token');
+                });
+
+            expect(regex.test(responseHeader)).to.be.false;
+            expect(requestHeaders).to.be.empty;
+        });
+    });
+
+    describe('custom headers are provided', function() {
+        before(function(done) {
+            this.run({
+                requester: {
+                    cacheControlHeader: false,
```

Comment:
these should be true to make sure it will not overwrite if true.

Reply:
shouldn't they be overwritten irrespective of true or false? since headers are in a request this would mean manually removing headers if options are true? I feel by default custom headers should have higher priority?